### PR TITLE
move createWindow to windowsReducer

### LINKED ISSUE #9962: window creation should be handled by windowsReducer
pre-req for other perf related changes

### PR DESCRIPTION
**Pull request by bridiver**
fix #9962

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/browser/reducers/windowsReducer.js [hunk 1]
```diff
@@ -4,18 +4,276 @@
...
+const firstDefinedValue = require('../../../js/lib/functional').firstDefinedValue
+const appConfig = require('../../../js/constants/appConfig')
+const messages = require('../../../js/constants/messages')
+const appUrlUtil = require('../../../js/lib/appUrlUtil')
+const settings = require('../../../js/constants/settings')
+const getSetting = require('../../../js/settings').getSetting
+const {zoomLevel} = require('../../common/constants/toolbarUserInterfaceScale')
+
+const isDarwin = process.platform === 'darwin'
+const isWindows = process.platform === 'win32'
```
Comment (NejcZdovc) [inline app/browser/reducers/windowsReducer.js]: Can we use `platformUtil` here?

File: app/browser/reducers/windowsReducer.js [hunk 2]
```diff
@@ -4,18 +4,276 @@
...
 const windows = require('../windows')
 const {makeImmutable} = require('../../common/state/immutableUtil')
+const electron = require('electron')
+const BrowserWindow = electron.BrowserWindow
+const firstDefinedValue = require('../../../js/lib/functional').firstDefinedValue
+const appConfig = require('../../../js/constants/appConfig')
+const messages = require('../../../js/constants/messages')
+const appUrlUtil = require('../../../js/lib/appUrlUtil')
+const settings = require('../../../js/constants/settings')
+const getSetting = require('../../../js/settings').getSetting
```
Comment (NejcZdovc) [inline app/browser/reducers/windowsReducer.js]: can we change to this `const {getSetting} = require('../../../js/settings')`?

File: app/browser/reducers/windowsReducer.js [hunk 3]
```diff
@@ -4,18 +4,276 @@
...
 
+const path = require('path')
 const appConstants = require('../../../js/constants/appConstants')
 const windowConstants = require('../../../js/constants/windowConstants')
 const windowState = require('../../common/state/windowState')
 const windows = require('../windows')
 const {makeImmutable} = require('../../common/state/immutableUtil')
+const electron = require('electron')
+const BrowserWindow = electron.BrowserWindow
+const firstDefinedValue = require('../../../js/lib/functional').firstDefinedValue
```
Comment (NejcZdovc) [inline app/browser/reducers/windowsReducer.js]: can we change to this `const {firstDefinedValue} = require('../../../js/lib/functional')`?

File: app/browser/reducers/windowsReducer.js [hunk 4]
```diff
@@ -4,18 +4,276 @@
...
+
+function windowDefaults (state) {
+  setDefaultWindowSize(state)
+
+  return {
+    show: false,
+    width: state.getIn(['defaultWindowParams', 'width']) || state.get('defaultWindowWidth'),
+    height: state.getIn(['defaultWindowParams', 'height']) || state.get('defaultWindowHeight'),
+    x: state.getIn(['defaultWindowParams', 'x']) || undefined,
+    y: state.getIn(['defaultWindowParams', 'y']) || undefined,
```
Comment (NejcZdovc) [inline app/browser/reducers/windowsReducer.js]: should we use `state.getIn(['defaultWindowParams', 'y'], undefined)` here? and for all examples above?

File: app/browser/reducers/windowsReducer.js (hunk 3 above)
Comment (bridiver) [inline app/browser/reducers/windowsReducer.js]: this is literally just a cut and paste from appStore to windowsReducer

File: app/browser/reducers/windowsReducer.js (hunk 2 above)
Comment (bridiver) [inline app/browser/reducers/windowsReducer.js]: same thing for all of these, this is just copied over and will go through a second round of refactoring to make other changes

**Comment by luixxiul** [comment]
this was moved to the latest milestone based on https://github.com/brave/browser-laptop/pull/9963#event-1159467263

File: app/browser/reducers/windowsReducer.js [hunk 5]
```diff
@@ -4,18 +4,276 @@
...
+function setDefaultWindowSize (state) {
+  if (!state) {
+    return
+  }
+  const screen = electron.screen
+  const primaryDisplay = screen.getPrimaryDisplay()
+  if (!state.getIn(['defaultWindowParams', 'width']) && !state.get('defaultWindowWidth') &&
+      !state.getIn(['defaultWindowParams', 'height']) && !state.get('defaultWindowHeight')) {
+    state = state.setIn(['defaultWindowParams', 'width'], primaryDisplay.workAreaSize.width)
+    state = state.setIn(['defaultWindowParams', 'height'], primaryDisplay.workAreaSize.height)
```
Comment (NejcZdovc) [inline app/browser/reducers/windowsReducer.js]: We need to return this state to the `appStore.js`

**Comment by luixxiul** [comment]
added to 0.20 branch?

**Comment by luixxiul** [comment]
moving the milestone in case. please revert it if it was confirmed that this should and was merged to 0.20, thanks

**Comment by NejcZdovc** [comment]
@luixxiul again please don't change milestones. This was merged to 0.20

**Pull request merged**